app/api/localStorage/index.js

```diff
@@ -262,4 +262,14 @@ export default class LocalStorageApi {
     await this.unsetHideBalance();
   }
 
+  setLocalStorage = (localStorageData: any): Promise<void> => new Promise((resolve) => {
+    Object.keys(localStorageData).forEach(key => {
+      localStorage.setItem(key, localStorageData[key]);
+    });
+    resolve();
```

Comment:
This looks buggy. If I remember correctly, `setItem` is async, so you can be calling resolve before it actually ends doing all the `setItem`

Reply:
I didn't find anything in the documentation, but several people reported this in StackOverflow. I'll add the `await`.

Reply:
I posted that from my phone. I looked it up and it’s `sync` so it should be okay.